src/kernel.ts

```diff
@@ -422,6 +459,9 @@ class Kernel implements IKernel {
    * received and validated.
    */
   kernelInfo(): Promise<IKernelInfo> {
+    if (this._status === KernelStatus.Dead) {
+      return Promise.reject(Error('Kernel is not ready to send a message'));
+    }
```

Comment:
This one could be, but the others have a more stringent definition of "ready to send".